module/Calendar.js

```diff
@@ -0,0 +1,311 @@
+	 * @param {function} [nextCallback]
+	 * Callback function that will be called when the specified date in the event occurs
+	 * @example
+	 * updateEvent('f9ca-1baa-c970-35b4', { title: 'birthday', date: '2022-04-06', time: '18:24:00'}, () => { console.log('Happy Birthday') })
+	 */
+	updateEvent(id, nextEvent, nextCallback) {
+		try {
+			const validatedEvent = this._eventValidator(nextEvent);
+			const validatedDate = this._dateValidator(nextEvent.date, nextEvent.time);
+			const events = this._events;
+			const index = events.findIndex((el) => el.id === id);
+			if (index !== -1) {
+				let callback;
+				if (!nextCallback) {
+					callback = events[index].callback;
+				} else {
+					callback = nextCallback;
+				}
+				const date = events[index].date + 'T' + events[index].time;
+				if (Date.parse(validatedDate) !== Date.parse(date)) {
+					clearTimeout(events[index].timerId);
+					const timerId = setTimeout(
+						callback,
+						new Date(validatedDate).getTime() - Date.now()
+					);
+					console.log(callback);
+					events[index] = {
+						...events[index],
+						...validatedEvent,
+						timerId,
+						callback,
+					};
+				}
+				events[index] = {
+					...events[index],
+					...validatedEvent,
+					callback,
+				};
+				return 'Updated successfully!';
+			} else {
+				throw new Error('Event not found!');
+			}
+		} catch (error) {
+			console.log(error.message);
+		}
+	}
+
+	/**
+	 * The method returns array with events objects in the specified date range
+	 * @method getEventsForPeriod
+	 * @param {string} dateRange1
+	 * @param {string} dateRange2
+	 * @example
+	 * getEventsForPeriod('2022-10-10', '2022-10-20')
+	 */
+	getEventsForPeriod(dateRange1, dateRange2) {
+		try {
+			if (!dateRange1 && !dateRange2) {
+				throw new Error('Date range is required!');
+			}
+			if (isNaN(Date.parse(dateRange1)) && isNaN(Date.parse(dateRange2))) {
+				throw new Error('Incorrect date format!');
+			}
+			const parsedDateRange1 = Date.parse(dateRange1);
+			const parsedDateRange2 = Date.parse(dateRange2);
+			const events = this._events.filter((event) => {
+				return (
+					(Date.parse(event.date) >= parsedDateRange1 &&
+						Date.parse(event.date) <= parsedDateRange2) ||
+					(Date.parse(event.date) >= parsedDateRange2 &&
+						Date.parse(event.date) <= parsedDateRange1)
+				);
+			});
+			if (events.length) {
+				return events;
+			} else {
+				return 'There are no events for the specified period!';
+			}
+		} catch (error) {
+			console.log(error.message);
+		}
+	}
+
+	/**
+	 * The method return object with event in the specified date
+	 * @method getEventForDay
+	 * @param {string} date
+	 * @example
+	 * getEventForDay('2022-10-10')
+	 */
+	getEventForDay(date) {
+		try {
+			if (!date) {
+				throw new Error('Date is required!');
+			}
+			if (isNaN(Date.parse(date))) {
+				throw new Error('Incorrect date format!');
+			}
+			const parsedDate = Date.parse(date);
```

Comment:
Duplicating code, calling `Date.parse(date)` twice. Take it to a variable

Reply:
fixed

Reply:
moved to variable